mirrord/layer/src/file/filter.rs

```diff
@@ -85,103 +97,154 @@ pub(crate) static FILE_FILTER: OnceLock<FileFilter> = OnceLock::new();
+            OldFilter::Include(include)
+        } else if !exclude.is_empty() {
+            let exclude = RegexSetBuilder::new(exclude)
+                .case_insensitive(true)
+                .build()
+                .expect("Building exclude regex set failed");
+            OldFilter::Exclude(exclude)
+        } else {
+            OldFilter::Nothing
         };
 
-        let exclude = reduce_to_string(exclude)
-            // Add default exclude list.
-            .map(|mut user_exclude| {
-                user_exclude.push_str(&format!("|{default_exclude}"));
-                user_exclude
-            })
-            .as_deref()
-            .map(Regex::new)
-            .transpose()
-            .expect("Failed parsing exclude file regex!")
-            // `exclude` always exists, either as user input, or as our default exclude list.
-            .unwrap_or(default_exclude_regex);
-
-        // Try to generate the final `Regex` based on `include`.
-        reduce_to_string(include)
-            .as_deref()
-            .map(Regex::new)
-            .transpose()
-            .expect("Failed parsing include file regex!")
-            .map(Self::Include)
-            // `include` was empty, so we fallback to `exclude`.
-            .unwrap_or(Self::Exclude(exclude))
+        Self {
+            old_filter,
+            read_only,
+            read_write,
+            local,
+            default_local,
+            mode,
+        }
     }
 
     /// Checks if `text` matches the regex held by the initialized variant of `FileFilter`,
-    /// converting the result a `Detour`.
+    /// and the whether the path is queried for write converting the result a `Detour`.
     ///
     /// `op` is used to lazily initialize a `Bypass` case.
-    pub(crate) fn continue_or_bypass_with<F>(&self, text: &str, op: F) -> Detour<()>
+    pub(crate) fn continue_or_bypass_with<F>(&self, text: &str, write: bool, op: F) -> Detour<()>
     where
         F: FnOnce() -> Bypass,
     {
         // Order matters here, as we want to make `include` the most important pattern. If the user
         // specified `include`, then we never want to accidentally allow other paths to pass this
         // check (this is a corner case that is unlikely to happen, as initialization via
         // `FileFilter::new` should prevent it from ever seeing the light of day).
-        match self {
-            FileFilter::Include(include) if include.is_match(text).unwrap() => Detour::Success(()),
-            FileFilter::Exclude(exclude) if !exclude.is_match(text).unwrap() => Detour::Success(()),
-            _ => Detour::Bypass(op()),
+        match &self.old_filter {
+            OldFilter::Include(include) if include.is_match(text) => {}
+            OldFilter::Exclude(exclude) if !exclude.is_match(text) => {}
+            OldFilter::Nothing => {}
+            _ => return Detour::Bypass(op()),
+        }
+
+        if some_regex_match!(&self.read_write, text) {
```

Comment:
What do you think of something like this (to drop the macro):

```suggestion
        if self.read_write.map(|regex| regex.is_match(text)).unwrap_or_default()
```

Reply:
Applied it as a macro (as it's still very repetitive)

Reply:
I would go for a closure instead of a macro there, as it is a small block of code.

The closure could take both the regex and `text` as references, and is clearer than a macro for this kind of stuff, imo.

Your approach is fine though, your pick.